decode.rb

```diff
@@ -0,0 +1,38 @@
+@decoder_letters = { '.-' => 'A', '-...' => 'B', '-.-.' => 'C', '-..' => 'D', '.' => 'E', '..-.' => 'F', '--.' => 'G',
+                     '....' => 'H', '..' => 'I', '.---' => 'J', '-.-' => 'K', '.-..' => 'L', '--' => 'M', '-.' => 'N',
+                     '---' => 'O', '.--.' => 'P', '--.-' => 'Q', '.-.' => 'R', '...' => 'S', '-' => 'T', '..-' => 'U',
+                     '...-' => 'V', '.--' => 'W', '-..-' => 'X', '-.--' => 'Y', '--..' => 'Z', '.----' => '1',
+                     '..---' => '2', '...--' => '3', '....-' => '4', '.....' => '5', '-....' => '6', '--...' => '7',
+                     '---..' => '8', '----.' => '9', '-----' => '0' }
+
+# method to decode a char
+def decode_char(string)
+  @decoder_letters.fetch(string)
+end
+
+puts decode_char('.-')
+
+# Method to decode the word
+def decode_word(string)
+  output = ''
+  string
+    .split
+    .each do |i|
+    output += decode_char(i)
+  end
+  output
+end
+
+puts decode_word('-- -.--')
+
+# Decoding a message
+def decode_message(_str)
+  output = []
+  string
+    .split('   ')
+    .each do |i|
+    output.push(decode_word(i))
+  end
+  output.join(' ').gsub(/\s+/, ' ')
+end
+puts decode_message('.-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...')
```

Comment:
- [ ] @myelin0, I think there's a problem with this method. It throws an error. Kindly make sure that the method is working as expected.
- **Screenshot:**
![Screenshot from 2022-07-20 18-18-17](https://user-images.githubusercontent.com/63087491/180043886-ab410658-50d6-4013-8a3d-011ec49b49af.png)



![](https://media.giphy.com/media/cMnt7i2RykmpW/giphy.gif)

Reply:
Thank you @FrederickMih for your review! I fixed the issue.